tests/integration/cloud/clouds/test_ec2.py

```diff
@@ -122,11 +123,10 @@ def test_instance_rename(self):
 
         changed_name = self.instance_name + '-changed'
 
-        rename_result = self.run_cloud(
-            '-a rename {0} newname={1} --assume-yes'.format(self.instance_name, changed_name), timeout=TIMEOUT)
-        self.assertFalse(self._instance_exists(), 'Instance wasn\'t renamed: |\n{}'.format(rename_result))
+        self.run_cloud('-a rename {0} newname={1} --assume-yes'.format(self.instance_name, changed_name), timeout=TIMEOUT)
+        sleep(20)
```

Comment:
I think that what I'd prefer to see here , rather than a simple `sleep(20)`, is do something like:

```
tries = 10
while tries > 0:
    tries -= 1
    if not self._instance_exists():
        tries = 0
    else:
        time.sleep(2)
```

Or something to that effect (that would wait for 20s total, max). One could increase the timeout length (e.g. 5s between checks) or whatever, to tune things. That would allow it to return faster if able, or take longer, if there are network issues.

Obviously one would want to have a reasonable maximum, and a reasonable time between checks...

I don't know if it's worth holding up the PR, but it might be a better choice.